src/point_one/fusion_engine/messages/fault_control.h

```diff
@@ -0,0 +1,195 @@
+/**************************************************************************/ /**
+ * @brief System fault control messages.
+ * @file
+ ******************************************************************************/
+
+#pragma once
+
+#include "point_one/fusion_engine/messages/defs.h"
+
+namespace point_one {
+namespace fusion_engine {
+namespace messages {
+
+// Enforce 4-byte alignment and packing of all data structures and values.
+// Floating point values are aligned on platforms that require it. This is done
+// with a combination of setting struct attributes, and manual alignment
+// within the definitions.
+#pragma pack(push, 1)
+
+/**
+ * @defgroup fault_control_messages System Fault Control
+ * @brief Messages/types for controlling or simulating system faults.
+ * @ingroup config_and_ctrl_messages
+ */
+
+/**
+ * @brief Available fault types/control inputs.
+ * @ingroup fault_control_messages
+ *
+ * See @ref FaultControlMessage.
+ */
+enum class FaultType : uint8_t {
+  /**
+   * Take no action.
+   *
+   * Payload format: none
+   */
+  NONE = 0,
```

Comment:
To be totally clear, sending a `FaultType::COCOM` type fault with `CoComType::NONE` clears the fault.

Does sending a `FaultType::NONE` clear a fault in any case, or does it truly act as a noop?

Reply:
On a related note, can for instance a COCOM, and GNSS fault run in parallel?

Reply:
No, it does not clear anything. I could just make it `RESERVED` or something?

Yes, you can do both COCOM and GNSS faults at the same time, though COCOM will stop the GNSS engine from updating so the GNSS disable wouldn't have any effect.

Reply:
Assuming `NONE` is a noop, It might be worth adding a `CLEAR_FAULTS` type.

Reply:
Yeah `NONE` isn't really useful. I'll just redefine it as `CLEAR_ALL`.